# Iterables and Iterators 

In [19]:
assign_txt="Narendra  Damodardas  Modi,  born  17  September  1950)  is  the  15th  and \
     current  Prime  Minister  of  India,  in  office  since  26  May  2014.odi,  a  leader \
of the Bharatiya Janata Party was the Chief Minister of Gujarat from 2001 to \
2014  and  is  the  Member  of  Parliament  from  Varanasi.  He  led  the  BJP  in  the \
2014  general  election,  which  gave  the  party  a  majority  in  the  Lok  Sabha,  the \
first for any political party in India since 1984.As the Chief Minister of \
Gujarat,  Modi's  economic  policies  were  praised,  while  his  administration  was \
also criticised for failing to significantly improve the human development \
in the state, and for failing to prevent the 2002 Gujarat riots. A Hindu \
nationalist and member of the Rashtriya Swayamsevak Sangh, Modi,remains a \
controversial figure domestically and internationally. Modi was born on 17 \
September  1950,  to  a  family  of  grocers  in  Vadnagar,  Mehsana  district,  Bombay \
State  (present-day  Gujarat).Modi's  family  belonged  to  the  Modh-Ghanchi-Teli \
(oil-presser) community,hich is categorised as an Other Backward Class by \
the  Indian  government.Modi  was  the  third  of  six  children  born  to  Damodardas \
Mulchand  (1915–1989)  and  Heeraben  Modi  (b.  c.  1920).As  a  child,  Modi  helped \
his father sell tea at the Vadnagar railway station, and later ran a tea \
stall with his brother near a bus terminus.Modi completed his higher \
secondary  education  in  Vadnagar  in  1967,  where  a  teacher  described  him  as  an \
average  student  and  a  keen  debater,  with  an  interest  in  theatre.Modi  had  an \
early gift for rhetoric in debates, and this was noted by his teachers and \
students.Modi  preferred  playing  larger-than-life  characters  in  theatrical \
productions,  which  has  influenced  his  political  image.At  age  eight,  Modi \
discovered  the  Rashtriya  Swayamsevak  Sangh  (RSS),  and  began  attending  its \
local shakhas (training sessions). There, Modi met Lakshmanrao Inamdar, \
popularly  known  as  Vakil  Saheb,  who  inducted  him  as  an  RSS  balswayamsevak \
(junior cadet) and became his political mentor.While Modi was training with \
the  RSS,  he  also  met  Vasant  Gajendragadkar  and  Nathalal  Jaghda,  Bharatiya \
Jana  Sangh  leaders  who  were  founding  members  of  the  BJP's  Gujarat  unit  in \
1980.Engaged  while  still  a  child  to  a  local  girl,  Jashodaben  Narendrabhai \
Modi,  Modi  rejected  the  arranged  marriage  at  the  same  time  he  graduated  from \
high  school.The  resulting  familial  tensions  contributed  to  his  decision  to \
leave home in 1967.Modi spent the ensuing two years travelling across \
Northern  and  North-eastern  India,  though  few  details  of  where  he  went  have \
emerged.In  interviews,  Modi  has  described  visiting  Hindu  ashrams  founded  by \
Swami  Vivekananda:  the  Belur  Math  near  Kolkata,  followed  by  the  Advaita \
Ashrama  in  Almora  and  the  Ramakrishna  mission  in  Rajkot.  Modi  remained  only \
a  short  time  at  each,  since  he  lacked  the  required  college  education."


An iterable is an object that can return an iterator. It is an object with an associated iter method. Once iter is applied to an iterable, an iterator object gets created. An iterator is an object that has an associated next method.

For example, Using a for loop, we can iterate over a list, range, dictionaries, file connections  or a string.

Any object that can be looped over is called as an iterable and it creates an iterator. Therefore we can iterate over an 
iterable

## Conditional interlude



In [1]:
# We can control the flow of execution of code by using conditionals such as if-else
if 2>3:#Notice that a statement is followed by a :
    print ('Is it possible?')
    
else:
    print ('Yay!!')

Yay!!


## Case Study


In a given text, 
- Count number of capital letters
- How many times has the word "RSS" occured?
- How many times has the word "Modi" occured?

In [2]:
assign_txt="Narendra Damodardas Modi, born 17 September 1950) is the 15th and current Prime Minister of India, in office since 26 May 2014.odi, a leader of the Bharatiya Janata Party was the Chief Minister of Gujarat from 2001 to 2014 and is the Member of Parliament from Varanasi. He led the BJP in the 2014 general election, which gave the party a majority in the Lok Sabha, the first for any political party in India since 1984.As the Chief Minister of Gujarat, Modi's economic policies were praised, while his administration was also criticised for failing to significantly improve the human development in the state, and for failing to prevent the 2002 Gujarat riots. A Hindu nationalist and member of the Rashtriya Swayamsevak Sangh, Modi,remains a controversial figure domestically and internationally. Modi was born on 17 September 1950, to a family of grocers in Vadnagar, Mehsana district, Bombay State (present-day Gujarat).Modi's family belonged to the Modh-Ghanchi-Teli (oil-presser) community,hich is categorised as an Other Backward Class by the Indian government.Modi was the third of six children born to Damodardas Mulchand (1915–1989) and Heeraben Modi (b. c. 1920).As a child, Modi helped his father sell tea at the Vadnagar railway station, and later ran a tea stall with his brother near a bus terminus.Modi completed his higher secondary education in Vadnagar in 1967, where a teacher described him as an average student and a keen debater, with an interest in theatre.Modi had an early gift for rhetoric in debates, and this was noted by his teachers and students.Modi preferred playing larger-than-life characters in theatrical productions, which has influenced his political image.At age eight, Modi discovered the Rashtriya Swayamsevak Sangh (RSS), and began attending its local shakhas (training sessions). There, Modi met Lakshmanrao Inamdar, popularly known as Vakil Saheb, who inducted him as an RSS balswayamsevak (junior cadet) and became his political mentor.While Modi was training with the RSS, he also met Vasant Gajendragadkar and Nathalal Jaghda, Bharatiya Jana Sangh leaders who were founding members of the BJP's Gujarat unit in 1980.Engaged while still a child to a local girl, Jashodaben Narendrabhai Modi, Modi rejected the arranged marriage at the same time he graduated from high school.The resulting familial tensions contributed to his decision to leave home in 1967.Modi spent the ensuing two years travelling across Northern and North-eastern India, though few details of where he went have emerged.In interviews, Modi has described visiting Hindu ashrams founded by Swami Vivekananda: the Belur Math near Kolkata, followed by the Advaita Ashrama in Almora and the Ramakrishna mission in Rajkot. Modi remained only a short time at each, since he lacked the required college education."

In [16]:
# .count method can be applied on the text to find the occurence of a particular word.
print (assign_txt.count('RSS'))

3


In [17]:
print (assign_txt.count('Modi'))

19


There are 19 occurrences of the word 'Modi' in the given text

In [5]:
type(assign_txt)

str

In [6]:
# To find the capital letters in the text, iterate through the same 
# and check if they are an upper case, if so, increment the counter by 1
c=0
for l in assign_txt:           
    if l.isupper():
        c = c+1
      
print (c)


120


There are 120 captial letters in the given text